docs/messaging/supplementary-guidelines/migration-guide.md

```diff
@@ -0,0 +1,434 @@
+| -------------- |
+| `messaging.batch.message_count` |
+| `messaging.destination.anonymous` |
+| `messaging.destination.template` |
+| `messaging.source.anonymous` |
+| `messaging.source.kind` |
+| `messaging.source.name` |
+| `messaging.source.template` |
+| `messaging.source.temporary` |
+<!-- prettier-ignore-end -->
+
+##### Changed attributes
+
+<!-- prettier-ignore-start -->
+| Attribute change | Comments |
+| ---------------- | -------- |
+| `messaging.destination` →  `messaging.destination.name` |  |
+| `messaging.destination_kind` →  `messaging.destination.kind` |  |
+| `messaging.temp_destination` →  `messaging.destination.temporary` |  |
+| `messaging.protocol` →  `net.app.protocol.name` |  |
+| `messaging.protocol_version` →  `net.app.protocol.version` |  |
+| `messaging.consumer_id` →  `messaging.consumer.id` |  |
+| `messaging.message_id` →  `messaging.message.id` |  |
+| `messaging.conversation_id` →  `messaging.message.conversation_id` |  |
+| `messaging.message_payload_size_bytes` →  `messaging.message.payload_size_bytes` |  |
+| `messaging.message_payload_compressed_size_bytes` →  `messaging.message.payload_compressed_size_bytes` |  |
+<!-- prettier-ignore-end -->
+
+##### Removed attributes
+
+<!-- prettier-ignore-start -->
+| Removed attributes |
+| -------------- |
+| `messaging.url` |
+<!-- prettier-ignore-end -->
+
+#### Kafka attributes
+
+##### New attributes
+
+<!-- prettier-ignore-start -->
+| New attributes |
+| -------------- |
+| `messaging.kafka.source.partition` |
+<!-- prettier-ignore-end -->
+
+##### Changed attributes
+
+<!-- prettier-ignore-start -->
+| Attribute change | Comments |
+| ---------------- | -------- |
+| `messaging.kafka.message_key` →  `messaging.kafka.message.key` |  |
+| `messaging.kafka.consumer_group` →  `messaging.kafka.consumer.group` |  |
+| `messaging.kafka.partition` →  `messaging.kafka.destination.partition` |  |
+| `messaging.kafka.tombstone` →  `messaging.kafka.message.tombstone` |  |
+<!-- prettier-ignore-end -->
+
+#### RabbitMQ attributes
+
+##### Changed attributes
+
+<!-- prettier-ignore-start -->
+| Attribute change | Comments |
+| ---------------- | -------- |
+| `messaging.rabbitmq.routing_key` →  `messaging.rabbitmq.destination.routing_key` |  |
+<!-- prettier-ignore-end -->
+
+#### RocketMQ attributes
+
+##### Changed attributes
+
+<!-- prettier-ignore-start -->
+| Attribute change | Comments |
+| ---------------- | -------- |
+| `messaging.rocketmq.message_type` →  `messaging.rocketmq.message.type` |  |
+| `messaging.rocketmq.message_tag` →  `messaging.rocketmq.message.tag` |  |
+| `messaging.rocketmq.message_keys` →  `messaging.rocketmq.message.keys` |  |
+| `messaging.rocketmq.delivery_timestamp` →  `messaging.rocketmq.message.delivery_timestamp` |  |
+| `messaging.rocketmq.delay_time_level` →  `messaging.rocketmq.message.delay_time_level` |  |
+| `messaging.rocketmq.message_group` →  `messaging.rocketmq.message.group` |  |
+<!-- prettier-ignore-end -->
```

Comment:
Looking into https://github.com/open-telemetry/semantic-conventions/pull/1090, it's be nice to mention that we renamed spans and defined a bunch of metrics.